presto-resource-group-managers/src/test/java/com/facebook/presto/resourceGroups/reloading/TestReloadingResourceGroupConfigurationManager.java

```diff
@@ -220,6 +224,42 @@ public void testRefreshInterval()
         manager.destroy();
     }
 
+    @Test
+    public void testSelectionContextMatch()
+    {
+        H2DaoProvider daoProvider = setup("test_selection_context");
+        H2ResourceGroupsDao dao = daoProvider.get();
+        dao.createResourceGroupsGlobalPropertiesTable();
+        dao.createResourceGroupsTable();
+        dao.createSelectorsTable();
+        dao.insertResourceGroup(1, "global", "1MB", 1000, 100, 100, "weighted", null, true, "1h", "1d", "1h", "1MB", "1h", null, ENVIRONMENT);
+        dao.insertResourceGroup(2, "sub1", "2MB", 4, 3, 3, null, 5, null, null, null, null, null, null, 1L, ENVIRONMENT);
+        dao.insertResourceGroup(3, "sub2", "4MB", 6, 3, 3, null, 5, null, null, null, "1h", "1MB", "2h", 1L, ENVIRONMENT);
+        dao.insertSelector(1, 1, "user1", null, null, null, null);
```

Comment:
you can remove this line - selectors should not send queries to non-leaf resource groups